contrib/platform/src/com/sun/jna/platform/win32/IPHlpAPI.java

```diff
@@ -264,6 +309,140 @@ public FIXED_INFO() {
         public int dwNumAddrs;
     }
 
+    /**
+     * Contains information that describes an IPv4 TCP connection.
+     */
+    @FieldOrder({ "dwState", "dwLocalAddr", "dwLocalPort", "dwRemoteAddr", "dwRemotePort", "dwOwningPid" })
+    class MIB_TCPROW_OWNER_PID extends Structure {
+        public int dwState;
+        public int dwLocalAddr;
+        public int dwLocalPort;
+        public int dwRemoteAddr;
+        public int dwRemotePort;
+        public int dwOwningPid;
+    }
+
+    /**
+     * Contains a table of IPv4 TCP connections on the local computer.
+     */
+    @FieldOrder({ "dwNumEntries", "table" })
+    class MIB_TCPTABLE_OWNER_PID extends Structure {
+        public int dwNumEntries;
+        public MIB_TCPROW_OWNER_PID[] table = new MIB_TCPROW_OWNER_PID[1];
+
+        public MIB_TCPTABLE_OWNER_PID(Pointer buf) {
+            super(buf);
+            // First element contains array size
+            this.dwNumEntries = buf.getInt(0);
+            if (this.dwNumEntries > 0) {
+                table = (MIB_TCPROW_OWNER_PID[]) new MIB_TCPROW_OWNER_PID().toArray(this.dwNumEntries);
+                read();
+            } else {
+                table = null;
+            }
+        }
+    }
+
+    /**
+     * Contains information that describes an IPv6 TCP connection.
+     */
+    @FieldOrder({ "LocalAddr", "dwLocalScopeId", "dwLocalPort", "RemoteAddr", "dwRemoteScopeId", "dwRemotePort",
+            "State", "dwOwningPid" })
+    class MIB_TCP6ROW_OWNER_PID extends Structure {
+        public byte[] LocalAddr = new byte[16];
+        public int dwLocalScopeId;
+        public int dwLocalPort;
+        public byte[] RemoteAddr = new byte[16];
+        public int dwRemoteScopeId;
+        public int dwRemotePort;
+        public int State;
+        public int dwOwningPid;
+    }
+
+    /**
+     * Contains a table of IPv6 TCP connections on the local computer.
+     */
+    @FieldOrder({ "dwNumEntries", "table" })
+    class MIB_TCP6TABLE_OWNER_PID extends Structure {
+        public int dwNumEntries;
+        public MIB_TCP6ROW_OWNER_PID[] table = new MIB_TCP6ROW_OWNER_PID[1];
+
+        public MIB_TCP6TABLE_OWNER_PID(Pointer buf) {
+            super(buf);
+            // First element contains array size
+            this.dwNumEntries = buf.getInt(0);
+            if (this.dwNumEntries > 0) {
+                table = (MIB_TCP6ROW_OWNER_PID[]) new MIB_TCP6ROW_OWNER_PID().toArray(this.dwNumEntries);
+                read();
+            } else {
+                table = null;
+            }
```

Comment:
I suggest to use the empty array for table (this would enable users to iterate directly over the array without a null check) and move this logic to the "read()" method. read coudl then be called from the constructor, if wanted. That would be IMHO more in line with normal structure calling conventions.

Reply:
Given that I do exactly that iteration in the test case, I agree.  Updated.